src/main/java/seedu/address/model/ItemModelManager.java

```diff
@@ -234,4 +253,72 @@ public void clear() {
     public void sort() {
         this.visualList = visualList.sort();
     }
+
+    private void togglePriorityMode() {
+        priorityMode = !priorityMode;
+    }
+
+    /**
+     * Enable and disable the priority mode
+     * @return a boolean value. If true, means priority mode is on, else returns false.
+     * @throws IllegalListException if the visualList is not a task list.
+     */
+    public boolean togglePriority() throws IllegalListException {
+        if (!(visualList instanceof TaskList)) {
+            throw new IllegalListException();
+        }
+
+        togglePriorityMode();
+        if (!priorityMode) {
+            sortedTask = null;
+            this.visualList = taskList;
+        } else {
+            sortedTask = new PriorityQueue<>((item1, item2) -> {
+                Task task1 = item1.getTask().get();
+                Task task2 = item2.getTask().get();
+                return task1.getPriority().compareTo(task2.getPriority());
+            });
+            for (int i = 0; i < taskList.size(); i++) {
+                Item item = taskList.get(i);
+                if (!item.getTask().get().isComplete()) {
+                    sortedTask.add(item);
+                }
+            }
+            this.visualList = getNextTask();
+        }
+        return priorityMode;
+    }
+
+    private VisualizeList getNextTask() {
+        TaskList result = new TaskList();
+        result.add(sortedTask.peek());
+        return result;
+    }
+
+    /**
+     * Mark an item with a task as done.
+     * @param index the index of the item to be marked as done.
+     * @return the item that is marked as done.
+     * @throws IllegalListException if the operation is not done on a task list.
+     */
+    public Item done(int index) throws IllegalListException {
```

Comment:
Do you want to name this method markComplete instead? So as to maintain consistency with the naming of markComplete method.

Reply:
ok that's fine with me